src/OrbitGl/App.h

```diff
@@ -558,7 +558,8 @@ class OrbitApp final : public DataViewFactory,
   // Only call from the capture thread
   void CaptureMetricProcessTimer(const orbit_client_protos::TimerInfo& timer);
 
-  void ShowHistogram(std::vector<uint64_t> data, const std::string& function_name) override;
+  void ShowHistogram(std::vector<uint64_t> data, const std::string& function_name,
+                     uint64_t function_id) override;
```

Comment:
The style guide bans default arguments on virtual functions.

Reply:
But this is not a virtual function.

Reply:
see in `AppInterface.h`

`virtual void ShowHistogram(std::vector<uint64_t> data, const std::string& function_name,
                             uint64_t function_id) = 0;`

Reply:
Yes, but here we are overwriting it.

Reply:
Anyway, maybe you are right. @beckerhe can I ask you for a third opinion?

Reply:
```
The style guide bans default arguments on virtual functions.
```

Yes please no default arguments on virtual functions. You have to put them in all places where you override the function and they can (and will) get out of sync, etc.

An alternative is overloads. You could have two versions of `ShowHistogram` but after looking at the use-case it would be clearer to make to have a `ShowHistogram` and `ClearHistogram` function. The latter would take no arguments at all.

Reply:
@beckerhe I've given that a thought, but t would require to define the ClearHistogram method in like 3-4 classes/interfaces. If you think that's the right thing to do, I'll do it, in another PR perhaps.

Reply:
Thanks for your patience :) @akopich @beckerhe